datastore-v1-proto-client/src/main/java/com/google/datastore/v1/client/RemoteRpc.java

```diff
@@ -81,7 +89,12 @@ public InputStream call(
       String methodName, MessageLite request, String projectId, String databaseId)
       throws DatastoreException {
     logger.fine("remote datastore call " + methodName);
-
+    Preconditions.checkArgument(
+        databaseIdsMatch(databaseId),
+        "database ids mismatched: request database id: "
+            + databaseId
+            + ", DatastoreOptions db: "
+            + this.databaseId);
```

Comment:
Should we make it work in the same way as in #1054 ?

In this case, A databaseId configured on query will be prioritised first, followed by the databaseId on the `DatastoreOptions`.

Reply:
At this point, the request is already built so we can't change the database id (unless we rebuilt it, which would make things a little messier because the request is a `MessageLite` type). Separately, the `google-cloud-datastore` module also behaves this way where it expects the db id to be the same as the DatastoreOptions. Let me know your thoughts.